tools/common/StaticRegistrar.cs

```diff
@@ -4001,6 +4054,145 @@ void Specialize (AutoIndentStringBuilder sb, ObjCMethod method, List<Exception>
 			}
 		}
 
+		MethodDefinition GetBlockWrapperCreator (ObjCMethod obj_method, int parameter)
+		{
+			// A mirror of this method is also implemented in Runtime:GetBlockWrapperCreator
+			// If this method is changed, that method will probably have to be updated too (tests!!!)
+			MethodDefinition method = obj_method.Method;
+			MethodDefinition first = method;
+			MethodDefinition last = null;
+			while (method != last) {
+				last = method;
+				var createMethod = GetBlockProxyAttributeMethod (method, parameter) ;
+				if (createMethod != null)
+					return createMethod;
+
+				method = GetBaseMethodInTypeHierarchy (method);
+			}
+
+			// Might be the implementation of an interface method, so find the corresponding
+			// MethodDefinition for the interface, and check for BlockProxy attributes there as well.
+			var map = PrepareMethodMapping (first.DeclaringType);
+			if (map != null && map.TryGetValue (first, out var list)) {
+				if (list.Count != 1)
+					throw Shared.GetMT4127 (first, list);
+				var createMethod = GetBlockProxyAttributeMethod (list [0], parameter);
+				if (createMethod != null)
+					return createMethod;
+			}
+
+			// Might be an implementation of an optional protocol member.
+			if (obj_method.DeclaringType.Protocols != null) {
+				foreach (var proto in obj_method.DeclaringType.Protocols) {
+					foreach (var pMethod in proto.Methods) {
+						if (!pMethod.IsOptional)
+							continue;
+						if (pMethod.Name != method.Name)
+							continue;
+						if (!TypeMatch (pMethod.ReturnType, method.ReturnType))
+							continue;
+						if (pMethod.Parameters.Length != method.Parameters.Count)
+							continue;
+						for (int i = 0; i < pMethod.Parameters.Length; i++)
+							if (!TypeMatch (pMethod.Parameters [i], method.Parameters [i].ParameterType))
+								continue;
```

Comment:
don't we have this matching code somewhere else ? I don't recall it (I know there were helpers methods in gendarme) but we compare methods in many places

Reply:
Yes, we have similar logic in other places, the problem is that here we're comparing instances of `MethodDefinition` and `ObjCMethod`, so those other places can't be used as-is because they're comparing two instances of the same type.